libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp

```diff
@@ -132,13 +148,54 @@ void GLBackend::init() {
         GL_GET_INTEGER(MAX_UNIFORM_BLOCK_SIZE);
         GL_GET_INTEGER(UNIFORM_BUFFER_OFFSET_ALIGNMENT);
 
+        if ( vendor.contains("NVIDIA") ) {
+            qCDebug(gpugllogging) << "NVIDIA card detected";
+            GL_GET_INTEGER(GPU_MEMORY_INFO_DEDICATED_VIDMEM_NVX);
+            GL_GET_INTEGER(GPU_MEMORY_INFO_TOTAL_AVAILABLE_MEMORY_NVX);
+            GL_GET_INTEGER(GPU_MEMORY_INFO_CURRENT_AVAILABLE_VIDMEM_NVX);
+
+            qCDebug(gpugllogging) << "GPU_MEMORY_INFO_DEDICATED_VIDMEM_NVX: " << GPU_MEMORY_INFO_DEDICATED_VIDMEM_NVX;
+            qCDebug(gpugllogging) << "GPU_MEMORY_INFO_TOTAL_AVAILABLE_MEMORY_NVX: " << GPU_MEMORY_INFO_TOTAL_AVAILABLE_MEMORY_NVX;
+            qCDebug(gpugllogging) << "GPU_MEMORY_INFO_CURRENT_AVAILABLE_VIDMEM_NVX: " << GPU_MEMORY_INFO_CURRENT_AVAILABLE_VIDMEM_NVX;
+            qCDebug(gpugllogging) << "sz: " << sizeof(_total_memory);
+
+            _total_memory = GPU_MEMORY_INFO_TOTAL_AVAILABLE_MEMORY_NVX * BYTES_PER_KIB;
+            _dedicated_memory = GPU_MEMORY_INFO_DEDICATED_VIDMEM_NVX * BYTES_PER_KIB;
+            _video_card = NVIDIA;
+
+
+        } else if ( vendor.contains("ATI")) {
+            qCDebug(gpugllogging) << "ATI card detected";
+            GL_GET_INTEGER(TEXTURE_FREE_MEMORY_ATI);
+
+            _total_memory = TEXTURE_FREE_MEMORY_ATI * BYTES_PER_KIB;
+            _dedicated_memory = _total_memory;
+            _video_card = ATI;
+        } else if ( vendor.contains("Intel")) {
+            unsigned int mem;
+
+           if ( ::gl::queryCurrentRendererIntegerMESA(GLX_RENDERER_VIDEO_MEMORY_MESA, &mem) ) {
```

Comment:
```suggestion
           if (::gl::queryCurrentRendererIntegerMESA(GLX_RENDERER_VIDEO_MEMORY_MESA, &mem)) {
```

Reply:
can you explain this?  right now it looks like this will only work on linux with Intel GPUs.  why not other OSes?

also should it set _video_card?

Reply:
I looked, so far I've not seen a way to do this for Intel other than on Linux. I got this from the glxinfo program.

Reply:
Update: Changed slightly in the last commit. Now the `queryCurrentRendererIntegerMESA` bit is done always when not NVIDIA or ATI, which should work better with the nouveau driver.

Unfortunately there doesn't seem to be a way to get the free VRAM.

Still Linux specific though, as so far I've not found a way to get this information on other systems. So on Windows/OSX this will fail and go to the last resort of using GPUIdent.